tools/rpi/hoymiles/__init__.py

```diff
@@ -482,21 +482,20 @@ def compose_esb_packet(packet, mtu=17, **params):
         fragment = compose_esb_fragment(packet[i:i+mtu], **params)
         yield fragment
 
-def compose_set_time_payload(timestamp=None):
+def compose_send_time_payload(cmdId):
     """
     Build set time request packet
 
-    :param timestamp: time to set (default: int(time.time()) )
-    :type timestamp: int
+    :param cmd to request
+    :type cmd: uint8
     :return: payload
     :rtype: bytes
     """
-    if not timestamp:
-        timestamp = int(time.time())
+    timestamp = int(time.time())
 
-    payload = b'\x0b\x00'
+    payload = struct.pack('>B', cmdId) + b'\x00'
     payload = payload + struct.pack('>L', timestamp)  # big-endian: msb at low address
-    payload = payload + b'\x00\x00\x00\x05\x00\x00\x00\x00'
+    payload = payload + b'\x00\x00\x00\x00\x00\x00\x00\x00'
```

Comment:
Here should the AlarmMessageId counter be inserted. This lets the Inverter know up till which AlarmId we have seen the log.

Reply:
I'm currently working on this one - I think tomorrow I can add the code to request the alarm request code. Needs some testing but the sun is already down for today.